src/prompt/engine.go

```diff
@@ -468,21 +468,19 @@ func New(flags *runtime.Flags) *Engine {
 
 	env.Init()
 	cfg := config.Load(env)
+	env.Var = cfg.Var
 
-	// load the template cache for extra prompts prior to
-	// rendering any template
-	if flags.Type == DEBUG ||
-		flags.Type == SECONDARY ||
-		flags.Type == TRANSIENT ||
-		flags.Type == VALID ||
-		flags.Type == ERROR {
-		env.LoadTemplateCache()
-	}
+	// To prevent cross-segment template referencing issues, this should not be moved elsewhere.
+	// Related: https://github.com/JanDeDobbeleer/oh-my-posh/discussions/2885#discussioncomment-4497439
+	env.PopulateTemplateCache()
```

Comment:
why don't we make this part of `env.Init()` in that case?

Reply:
@JanDeDobbeleer I tried that, but it causes a problem. The order of these calls should not be changed:

https://github.com/JanDeDobbeleer/oh-my-posh/blob/388fa2e128def345a073990e239815c33637c4ea/src/prompt/engine.go#L469-L471

The config have to be loaded after `env.Init()` so that `env.Var` can be properly assigned and then be passed to the template cache:

https://github.com/JanDeDobbeleer/oh-my-posh/blob/388fa2e128def345a073990e239815c33637c4ea/src/runtime/terminal.go#L641-L643

Reply:
I'm refactoring on top of your HEAD as it occurred to me `TemplateCache` should not be part of `runtime`, but `template`. This also solves this remark, as it can be part of `template.Init` instead. It shouldn't have any side effects but it's quite a refactor so give me tomorrow as well to get it right.

Reply:
Still on it, almost there. I have one issue I need to fix and then I can push this change on top.

Reply:
I'm so sorry, this went in a whole new direction 😓 But I do feel this is a lot cleaner.

Reply:
Never mind. I'll find a moment and have a look.

Reply:
Seems to be good.

Reply:
I'm finishing this today. I'm going to fix the linter issues and do a final validation.